pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -276,6 +276,67 @@ func (lb *ListenerBuilder) inboundChainForOpts(cc inboundChainConfig, mtls authn
 	return chains
 }
 
+func getSidecarIngressPortList(node *model.Proxy) map[int]bool {
+	sidecarScope := node.SidecarScope
+	ingressPortListMap := make(map[int]bool)
+	for _, ingressListener := range sidecarScope.Sidecar.Ingress {
+		ingressPortListMap[int(ingressListener.Port.Number)] = true
+	}
+	return ingressPortListMap
+}
+
+func (lb *ListenerBuilder) getFilterChainsByServicePort(chainsByPort map[uint32]inboundChainConfig,
+	enableSidecarServiceInboundListenerMerge bool,
+) map[uint32]inboundChainConfig {
+	ingressPortListMap := make(map[int]bool)
+	sidecarScope := lb.node.SidecarScope
+	if sidecarScope.HasIngressListener() {
+		ingressPortListMap = getSidecarIngressPortList(lb.node)
+	}
+	for _, i := range lb.node.ServiceInstances {
+		port := ServiceInstancePort{
+			Name:       i.ServicePort.Name,
+			Port:       uint32(i.ServicePort.Port),
+			TargetPort: i.Endpoint.EndpointPort,
+			Protocol:   i.ServicePort.Protocol,
+		}
+		actualWildcards, _ := getWildcardsAndLocalHostForDualStack(lb.node.GetIPMode())
+		if enableSidecarServiceInboundListenerMerge && sidecarScope.HasIngressListener() {
+			// here if port is declared in service and sidecar ingress both, we continue to take the one on sidecar + other service ports
+			// e.g. 1,2, 3 in service and 3,4 in sidecar ingress,
+			// this will still generate listeners for 1,2,3,4 where 3 is picked from sidecar ingress
+			if _, ok := ingressPortListMap[int(port.Port)]; ok {
+				// port present in sidecarIngress listener so let sidecar take precedence
+				continue
+			}
+		}
+		cc := inboundChainConfig{
```

Comment:
rename cc to a more accurate name

Reply:
this is already in existing code so kept it vs changing it to confuse!